src/realm/obj.cpp

```diff
@@ -687,13 +687,23 @@ void Obj::traverse_path(Visitor v, PathSizer ps, size_t path_length) const
                 TableRef tr = m_table->get_opposite_table(col_key);
                 Obj obj = tr->get_object(backlinks[0]); // always the first (and only)
                 auto next_col_key = m_table->get_opposite_column(col_key);
-                size_t index = 0;
+                Mixed index;
```

Comment:
`index` seems like the wrong name now?

Reply:
Well - it is still an index. Can't you say that you index the dictionary by the key?

Reply:
I would prefer `util::Optional<size_t> index` or even just using `npos`, as is the pattern many other places.

Reply:
I had misunderstood the code, resolving.